pkg/hintrunner/hint.go

```diff
@@ -906,3 +906,113 @@ func (hint *GetNextDictKey) Execute(vm *VM.VirtualMachine, ctx *HintRunnerContex
 }
+
+type AssertLeFindSmallArc struct {
+	a             ResOperander
+	b             ResOperander
+	rangeCheckPtr ResOperander
+}
+
+func (hint AssertLeFindSmallArc) String() string {
+	return "AssertLeFindSmallArc"
+}
+
+func (hint AssertLeFindSmallArc) Execute(vm *VM.VirtualMachine, ctx *HintRunnerContext) error {
+	primeOver3High, err := uint256.FromDecimal("3544607988759775765608368578435044694")
+	if err != nil {
+		return err
+	}
+	primeOver2High, err := uint256.FromDecimal("5316911983139663648412552867652567041")
+	if err != nil {
+		return err
+	}
+
+	a, err := hint.a.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve a operand %s: %v", hint.a, err)
+	}
+
+	b, err := hint.b.Resolve(vm)
+	if err != nil {
+		return fmt.Errorf("resolve b operand %s: %v", hint.b, err)
+	}
+
+	aFelt, err := a.FieldElement()
+	if err != nil {
+		return err
+	}
+
+	bFelt, err := b.FieldElement()
+	if err != nil {
+		return err
+	}
+
+	thirdLength := f.Element{32, 0, 0, 544} // -1 field element
+	thirdLength.Sub(&thirdLength, bFelt)
+
+	// Array of pairs (2-tuple)
+	lengthsAndIndices := []struct {
+		Value    f.Element
+		Position int
+	}{
+		{*aFelt, 0},
+		{*bFelt.Sub(bFelt, aFelt), 1},
+		{thirdLength, 2},
+	}
+
+	// Sort
+	sort.Slice(lengthsAndIndices, func(i, j int) bool {
+		// return lengthsAndIndices[i].Value < lengthsAndIndices[j].Value
+		return lengthsAndIndices[i].Value.Cmp(&lengthsAndIndices[j].Value) == -1
+	})
+
+	// Exclude the largest arc after sorting
+	ctx.ExcludedArc = lengthsAndIndices[2].Position
+
+	// Find the modulo of the first element's value of the sorted array
+	// w.r.t. primeOver3High
+	tmp := uint256.Int(lengthsAndIndices[0].Value.Bits())
+	tmp.Mod(&tmp, primeOver3High)
+	val1 := f.Element{}
+	val1.SetBytes(tmp.Bytes())
+
+	// Find the modulo of the second element' value of the sorted array
+	// w.r.t. primeOver2High
+	tmp = uint256.Int(lengthsAndIndices[1].Value.Bits())
+	tmp.Mod(&tmp, primeOver2High)
+	val2 := f.Element{}
+	val2.SetBytes(tmp.Bytes())
+
+	rangeCheckPtrMemAddr, err := ResolveAsAddress(vm, hint.rangeCheckPtr)
+
+	if err != nil {
+		return fmt.Errorf("resolve rangeCheckPtr cell addr: %v", err)
+	}
+
+	writeValue := mem.MemoryValueFromFieldElement(&val1)
+	err = vm.Memory.WriteToAddress(&rangeCheckPtrMemAddr, &writeValue)
+	if err != nil {
+		return fmt.Errorf("write rangeCheckPtr cell: %v", err)
+	}
+
+	rangeCheckPtrMemAddr.Offset += 1
+	err = vm.Memory.WriteToAddress(&rangeCheckPtrMemAddr, &writeValue)
+	if err != nil {
+		return fmt.Errorf("write rangeCheckPtr+1 cell: %v", err)
+	}
+
+	rangeCheckPtrMemAddr.Offset += 1
+	writeValue = mem.MemoryValueFromFieldElement(&val2)
+	err = vm.Memory.WriteToAddress(&rangeCheckPtrMemAddr, &writeValue)
+	if err != nil {
+		return fmt.Errorf("write rangeCheckPtr+2 cell: %v", err)
+	}
+
+	rangeCheckPtrMemAddr.Offset += 1
+	err = vm.Memory.WriteToAddress(&rangeCheckPtrMemAddr, &writeValue)
+	if err != nil {
+		return fmt.Errorf("write rangeCheckPtr+3 cell: %v", err)
+	}
```

Comment:
Why are you storing the same value twice?

Reply:
From what I see on those files, it is first storing the remainder and then the result of the division